### PR TITLE
Prevent GCalendar events from creating duplicates

### PR DESCRIPTION
**Pull request by bordoni**
🎫 [TEC-4497]
---
Kept creating new instances until the new code.

Screen Shot 2022-09-29 at 8 36 05 AM

Kinda reverts code from #3966

[TEC-4497]: https://theeventscalendar.atlassian.net/browse/TEC-4497?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by lucatume — COMMENTED** [review]
Since it was broken and no one noticed and it’s now fixed and no one noticed, I would add tests on the `Aggregator/Record/Abstract` object.
Check the `wpunit` suite failure.

From the PR description, it's not clear to me what the issue was and, thus, what the fix is.

**Review by lucatume — CHANGES_REQUESTED** [review]
Add tests to cover the fix from future regression.

**Comment by bordoni** [comment]
Added tests

File: tests/wpunit/Tribe/Events/Aggregator/Record/AbstractTest.php [hunk 1]
```diff
@@ -282,6 +281,69 @@ public function should_correctly_insert_posts() {
 		$this->assertEquals( 0, $activity->count( Main::POSTTYPE,'skipped' ) );
 	}
 
+	/**
+	 * It should correctly insert posts
```
Comment (lucatume) [inline tests/wpunit/Tribe/Events/Aggregator/Record/AbstractTest.php:285]: Copy and pasta here?

File: tests/wpunit/Tribe/Events/Aggregator/Record/AbstractTest.php [hunk 2]
```diff
@@ -282,6 +281,69 @@ public function should_correctly_insert_posts() {
...
+			]
+		] );
+
+		$manual_test_record = new \Tribe__Events__Aggregator__Record__Manual_Test( $record );
+		$manual_test_record->meta['origin'] = 'manual-test';
+
+		// Enforce this particular Static variable.
+		\Tribe__Events__Aggregator__Record__Manual_Test::$unique_id_fields = \Tribe__Events__Aggregator__Record__Abstract::$unique_id_fields;
+
+		$data = [
```
Comment (lucatume) [inline tests/wpunit/Tribe/Events/Aggregator/Record/AbstractTest.php:335]: Formatting a bit wonky here.

File: tests/wpunit/Tribe/Events/Aggregator/Record/AbstractTest.php [hunk 3]
```diff
@@ -282,6 +281,69 @@ public function should_correctly_insert_posts() {
...
+			[
+				'uid' => $uid,
```
Comment (lucatume) [inline tests/wpunit/Tribe/Events/Aggregator/Record/AbstractTest.php:337]: Adding, either here or in a dedicated test, a `uid` pointing to an event that is not existing, would test the mixed, and more realistic nature, of this code.

**Pull request merged**